### DISCUSSION TITLE
Assertion error: something relative to module resolution

### DISCUSSION THREAD

**Discussion by SimTheFool** [avajs/ava#3120] [Q&A]
Hello !

I'm trying to launch tests on some pre-compiled and bundled test files (I know it's unusual but I needed that way). I'm building my project with esbuild and I'm on `"ava": "^4.3.3"`. I've set my package.json type to module, for convenience. I'm on `node v16`.

When launching the compiled tests, I'm having some issue where node is telling me I should use .cjs file if I want to resolve `require("path")` used by ava. But when I do so (changing my compiled files to `.cjs`, I get the following error:

```
task: [test] yarn ava ./**/***.spec.cjs
yarn run v1.22.18
$ /var/www/html/FellSeal-Builder/node_modules/.bin/ava './**/***.spec.cjs'

  Uncaught exception in build-test/domain/character/buildCharacter.spec.cjs

  build-test/domain/character/buildCharacter.spec.cjs:93

   92:     var { flags, refs } = require_state();
   93:     assert(refs.runnerChain);             
   94:     flags.loadedMain = true;              

  AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:

    assert(refs.runnerChain)


  › assert(refs.runnerChain)
  › __require (build-test/domain/character/buildCharacter.spec.cjs:26:50)
  › Object.<anonymous> (build-test/domain/character/buildCharacter.spec.cjs:100:27)

  ✖ build-test/domain/character/buildCharacter.spec.cjs exited with a non-zero exit code: 1
  ─

  1 uncaught exception
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Command failed: node /home/simthefool/.yvm/versions/v1.22.18/bin/yarn.js ava ./**/***.spec.cjs


```

So I tried another way. I'm keeping my extension to `.js`, but I'm erasing the default `require` at compile time, to get it from `import`. The still issue persists.

--- Additional infos ---
Here is my esbuild config in case the issu comes from compilation. I'm switching the two last entries to try both solution I explained.

```
{
    entryPoints: entries,
    outdir: outDir,
    outbase: "./src",
    bundle: true,
    platform: "node",
    /* outExtension: { '.js': '.cjs' }, */
    banner: {
      js: "import { createRequire as topLevelCreateRequire } from 'module';\n const require = topLevelCreateRequire(import.meta.url);",
    },
  }
```
And here is my package.json:
```
{
  "name": "builder",
  "license": "UNLICENSED",
  "type": "module",
  "version": "1.0.0",
  "dependencies": {
    "lodash": "^4.17.21",
    "utils": "*",
    "uuid": "^9.0.0"
  },
  "devDependencies": {
    "@types/lodash": "^4.14.186",
    "@types/node": "^18.7.23",
    "@types/uuid": "^8.3.4",
    "ava": "^4.3.3",
    "esbuild": "^0.15.11",
    "tiny-glob": "^0.2.9",
    "typescript": "^4.8.4",
    "yargs": "^17.6.0"
  }
}

```

Also, I'm using yarn workspace. Here is my repo: [https://github.com/SimTheFool/FellSeal-Builder](https://github.com/SimTheFool/FellSeal-Builder). (I created a specific branch issue/ava-on-compiled-tests-01 so I can continue s I can continue to work without changing the specific repo state that triggered the issue).

Thanks for any help you can provide ;)

**Comment by novemberborn** [comment]
It looks like a copy of AVA may be included in your build. That doesn't work, the version of AVA imported by your test file needs to be the same as what the AVA CLI loaded into the worker thread / child process.

**Reply by SimTheFool** [reply]
Ok ! I don't really understand what I'm doing wrong. I've only one version of AVA in my node_modules. Isn't the version of AVA included in my test files the same as the version of the AVA CLI called with the command `yarn ava` ? If so, I don't think it's an inconsistent version issue.

**Reply by novemberborn** [reply]
Your build process may be creating a copy. Leave the tests out of the build and import the resulting bundle itself if possible.
